### ISSUE TITLE
i18n support and add Chinese document

### ISSUE THREAD

**Issue by nkwangleiGIT** [ray-project/ray#48523]
### Description

hi, I'd like to contribute the Chinese version of Ray documention, and also add the i18n for Ray doc using [Sphinx i18n](https://www.sphinx-doc.org/en/master/usage/advanced/intl.html) , here is my plan that need your suggstion about how to move it forward.

1. Add i18n to https://github.com/ray-project/ray/tree/master/doc/source, and it'll add a locale folder and also zh_CH for Chinese translation.
2. Translate some pages to Chinese at the 1st commit as baseline, including the nav, homepage, overview and getting started pages
3. Add a guide about how to translate more pages based on this commit
4. Add more translated Chinese pages piece by piece at latter commit, and others can also contribute new languages.

Some questions here：
1. seems sphinx doesn't support multi-language switch, check [similiar issue](https://github.com/sphinx-doc/sphinx/issues/788), I think we have to build each language separately, any other idea here?
2. how Ray document is built for now? we need add Chinese document build to the pipeline and release it together with English one, maybe we can host all languages behind a proxy and use different context to visit the expected language? such as /en for English, /zh_CH for Chinese, etc...

Appreciate your suggstions, thanks.

**Comment by Superskyyy** [comment]
Count me in if we ever seriously start on this path, translation is not a minor effort though even with AI's help. We need to keep it up, especially using a platform to track updates, docs are being modified in English at any time of the day.

**Comment by nkwangleiGIT** [comment]
yes, plan to contriube some time for Ray's Chinese translation, as there are more and more Chinese users to use Ray, I think it'll be great to help them get started easily with Ray if there are Chinese document.
I'm not sure if there is any organization in Transifex, so we can collaborate using this service. About how to sync with English version, I think it'll be a long story.
@Superskyyy  sure, I'll bring you to the loop if get some progress.

**Comment by jcotant1** [comment]
Cc: @angelinalg for visibility

**Comment by Superskyyy** [comment]
> yes, plan to contriube some time for Ray's Chinese translation, as there are more and more Chinese users to use Ray, I think it'll be great to help them get started easily with Ray if there are Chinese document. I'm not sure if there is any organization in Transifex, so we can collaborate using this service. About how to sync with English version, I think it'll be a long story. @Superskyyy sure, I'll bring you to the loop if get some progress.

So basically a GitHub action is necessary to sync English docs to Transifex, whenever a commit is merged into master, then someone can volunteer to work on the Chinese part, the untranslated will remain in English. I think this way we can achieve a proper sync, doing the work without sync is a huge pain in my experience.

**Comment by Superskyyy** [comment]
> yes, plan to contriube some time for Ray's Chinese translation, as there are more and more Chinese users to use Ray, I think it'll be great to help them get started easily with Ray if there are Chinese document. I'm not sure if there is any organization in Transifex, so we can collaborate using this service. About how to sync with English version, I think it'll be a long story. @Superskyyy sure, I'll bring you to the loop if get some progress.

Would you mind connecting with me on Wechat? I will send you an email later.

**Comment by nkwangleiGIT** [comment]
> Would you mind connecting with me on Wechat? I will send you an email later.

np, just send me your wechat to nkwanglei@126.com, I'll add you, thanks.

**Comment by Superskyyy** [comment]
> > Would you mind connecting with me on Wechat? I will send you an email later.
> 
> np, just send me your wechat to [nkwanglei@126.com](mailto:nkwanglei@126.com), I'll add you, thanks.

Sent, thanks.

**Comment by hwhsu1231** [comment]
Hello, @nkwangleiGIT @Superskyyy 

Regarding the creation of the translation project for the Ray Documentation: since the Ray Documentation is based on Sphinx, I believe the infrastructure of the @localizethedocs organization could be a great help. You can take a look at the following projects as examples:

- https://github.com/localizethedocs/cmake-docs-l10n
- https://github.com/localizethedocs/conan-docs-l10n
- https://github.com/localizethedocs/linux-docs-l10n
- https://github.com/localizethedocs/ros2-docs-l10n
- https://github.com/localizethedocs/llvm-docs-l10n

I have added the [ray-docs-l10n](https://github.com/localizethedocs/localizethedocs.github.io/issues/491) project to the TODO list. Once the project completes its incubation phase, I will make sure to notify you. In the meantime, I was wondering if you would be interested in contributing to the Simplified Chinese translations for other @localizethedocs projects?

**Comment by dstrodtman** [comment]
Closing out this ticket as stale.

Docs team recommends using browser-native machine translation at this time, pending future investment in publishing curated translations.
